ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/AttestationValidator.java

```diff
@@ -130,6 +132,20 @@ SafeFuture<InternalValidationResultWithState> singleOrAggregateAttestationChecks
                 return completedFuture(InternalValidationResultWithState.ignore());
               }
               final BeaconState state = maybeState.get();
+
+              final Optional<SlotInclusionGossipValidationResult>
+                  slotInclusionGossipValidationWithStateResult =
+                      attestationUtil.performSlotInclusionGossipValidation(attestation, state);
```

Comment:
If we were going to do this, we'd be needing to pass the head state, not just any state chosen for gossip topic inclusion. Head state is also sketchy (eg. future slot errors where we haven't completed epoch transitions).

Because it's using the state to compute the current epoch and previous epoch, if we end up using something like a fork state from previous epoch as the best state, then we'd actually be using current-1, current-2, and that'd be bad...
My suggestion if they're only using state to check against epoch is we compute the epoch against wall time inside that function, and then even if the state machine is behind we're accepting gossip correctly...

Then `performSlotInclusionGossipValidation` would only take the attestation... unless we need a state because of test cases or something...

If we did we could probably just have a wrapper interface that takes attestation and state and computes the epoch of the state then calls a private function that takes `attestation, headEpoch`, and the default public interface could derive headEpoch from wallTime...

Anyway, that's a couple of ideas. happy to talk further.